File: core/src/com/alta_v2/provider/AppConfigProvider.java
package com.alta_v2.provider;

import com.alta_v2.game.utils.Resources;
import com.alta_v2.rendering.config.AppConfig;
import com.google.inject.Provider;
import org.yaml.snakeyaml.Yaml;
import org.yaml.snakeyaml.constructor.Constructor;

import java.io.InputStream;

public class AppConfigProvider implements Provider<AppConfig> {
    @Override
    public AppConfig get() {
        Yaml yaml = new Yaml(new Constructor(AppConfig.class));
        InputStream inputStream = this.getClass()
                .getClassLoader()
                .getResourceAsStream(Resources.APP_CONFIG);
        return yaml.load(inputStream);
    }
}

File: core/src/com/alta_v2/provider/AppConfigProvider.kt
package com.alta_v2.provider

import com.alta_v2.game.utils.Resources
import com.alta_v2.rendering.config.AppConfig
import com.google.inject.Provider
import org.yaml.snakeyaml.Yaml
import org.yaml.snakeyaml.constructor.Constructor

class AppConfigProvider : Provider<AppConfig> {

    override fun get(): AppConfig {
        val yaml = Yaml()
        val inputStream = this.javaClass
                .classLoader
                .getResourceAsStream(Resources.APP_CONFIG)
        return yaml.loadAs(inputStream, AppConfig::class.java)
    }
}